src/rest.cpp

```diff
@@ -133,25 +133,28 @@ static ChainstateManager* GetChainman(const std::any& context, HTTPRequest* req)
     return node_context->chainman.get();
 }
 
-static RESTResponseFormat ParseDataFormat(std::string& param, const std::string& strReq)
+RESTResponseFormat ParseDataFormat(std::string& param, const std::string& strReq)
 {
-    const std::string::size_type pos = strReq.rfind('.');
-    if (pos == std::string::npos)
-    {
-        param = strReq;
-        return rf_names[0].rf;
-    }
+    // Remove query string (if any, separated with '?') as it should not interfere with
+    // parsing param and data format
+    param = strReq.substr(0, strReq.rfind('?'));
+    const std::string::size_type pos_format = param.rfind('.');
 
-    param = strReq.substr(0, pos);
-    const std::string suff(strReq, pos + 1);
+    // No format string is found
+    if (pos_format == std::string::npos)
+        return rf_names[0].rf;
 
+    // Match format string to available formats
+    auto len_suff = param.size() - pos_format - 1;  // - 1 so we don't include the leading .
+    const std::string suff(param, pos_format + 1, len_suff);
```

Comment:
Calculating the length shouldn't be necessary here, you can simply omit the third parameter of the ctor to get the sub-string from pos_format + 1 to the end:
```suggestion
    const std::string suff(param, pos_format + 1);
```
(I checked that tests still pass after this change)
Also, I'd suggest to rename "suff" to "suffix" to make it more clear (nit though)

Reply:
Ah yes, you're right, that is more elegant. Updated to your suggestion, and also renamed `suff` to `suffix`.